sdk/textanalytics/ai-text-analytics/src/util.ts

```diff
@@ -16,14 +25,20 @@ export function sortByPreviousIdOrder<T extends IdObject, U extends IdObject>(
   sortedArray: T[],
   unsortedArray: U[]
 ): U[] {
-  if (sortedArray.length !== unsortedArray.length) {
-    throw new Error("Can't use this sort unless you have two arrays with the same length!");
-  }
-
   const unsortedMap = new Map<string, U>();
   for (const item of unsortedArray) {
+    if (isGenericError(item)) {
+      throw new RestError(item.error.message, item.error.code, 400);
+    }
     unsortedMap.set(item.id, item);
   }
 
+  if (unsortedArray.length !== sortedArray.length) {
+    const ordinal = unsortedArray.length > sortedArray.length ? "more" : "fewer";
+    logger.warning(
+      `The service returned ${ordinal} responses than inputs. Some errors may be treated as fatal. Proceeding.`
```

Comment:
nit: is proceeding necessary here? I feel like warnings are understood to be non-fatal.